nixos/modules/services/networking/nomad.nix

```diff
@@ -85,10 +96,16 @@ in
         StateDirectory = "nomad";
         TasksMax = "infinity";
         User = optionalString cfg.dropPrivileges "nomad";
+      } // (if !cfg.enableDocker then { } else {
```

Comment:
I think you can use `lib.optionalAttrs` here:

```nix
// lib.optionalAttrs cfg.enableDocker { SupplementaryGroups = "docker"; }
```

Reply:
Or
```
serviceConfig = mkMerge [
  { DynamicUser = cfg.dropPrivileges;
    ...
  }
  (mkIf cfg.enableDocker { SupplementaryGrousp = "docker"; })
];
```

Also, isn't `SupplementaryGroups` a `list` (conceptually)?